Improve automated tests

scytacki opened the pull request:
This PR:
- refactors some of the cypress tests to remove redundant code
- fixes an issue with the travis config that was preventing it from sending the code coverage to code climate.
- disables firestore networking when the demo report is loaded
- fixes some cypress tests that were previously skipped because they were failing
- updates some cypress tests now that the user settings are starting empty on each test run

The most invasive change is the disabling of the firestore networking. This should only affect the demo report: the report that loads when you aren't coming from the portal.  Now the demo report won't remember any changes you made, but within the current tab session it should work as normal.

knowuh commented on cypress/utils.js:
```diff
@@ -1,9 +1,91 @@
-  getByCypressTag: function( cypressTag ) {
-    return cy.get(`[data-cy=${cypressTag}]`);
-  },
+  getByCypressTag: getByCypressTag,
```
I think a valid simplification here is `module.exports = {getByCypressTag, getAnswerByQuestionType, getPageQuestionData, ... etc}`  

[Object Literal Property Value Shorthand](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand)

knowuh commented on cypress/utils.js:
```diff
@@ -1,9 +1,91 @@
+  getSectionData:  getSectionData,
+  getPageData:  getPageData,
+  getActivityQuestionData: getActivityQuestionData
 };
```
I think a valid simplification here is `module.exports = {getByCypressTag, getAnswerByQuestionType, getPageQuestionData, ... etc}`  

[Object Literal Property Value Shorthand](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand)

scytacki commented:
Thanks @knowuh for reviewing. I think I addressed all of your comments.

I'm not sure if the tests run faster. I've been using the code coverage support which makes them run slower. And I've been adding tests so that is changing the total test time.  My sense is that they run a little faster.
